### DISCUSSION TITLE
amdsmi_get_energy_count() returns the same value

### DISCUSSION THREAD

**Discussion by parthraut** [ROCm/ROCm#3044] [Q&A] [answered]
Hello! So I am currently involved in bringing AMD GPU support to the [Zeus Project](https://ml.energy/zeus/).

We are using ROCM 6.0.2 and testing on [AMD HPC Fund ](https://amdresearch.github.io/hpcfund/) Cluster, specifically on an MI100.

We are using [amdsmi_get_energy_count()](https://rocm.docs.amd.com/projects/amdsmi/en/latest/py-interface_readme_link.html#amdsmi-get-energy-count) to query the energy consumption of the GPU while training. However, the values returned seem to not be updating, resulting in no change in energy consumption, as shown below:

```
>>> import amdsmi
>>> amdsmi.amdsmi_init()
>>> handles = amdsmi.amdsmi_get_processor_handles()
>>> amdsmi.amdsmi_get_energy_count(handles[0])
{'power': 8152136, 'counter_resolution': 15.300000190734863, 'timestamp': 3824190842400799}
>>> amdsmi.amdsmi_get_energy_count(handles[0])
{'power': 8152136, 'counter_resolution': 15.300000190734863, 'timestamp': 3824190842400799} # returns the same power value
```

Even while fully utilizing the GPU during a pytorch training script, there seems to be no difference in the return value, resulting in the calculated energy consumption to be 0 always. 

Is there something else that needs to be done for it to return an updated value?

**Comment by huanrwan-amd** [comment]
Hi @parthraut, sorry for the late reply. Can you please update ROCm software stack to the latest (Now is ROCm6.2.4, https://rocm.docs.amd.com/en/latest/) and try 
`watch -n 0.01 rocm-smi`
to check whether rocm-smi can give the power consumption value? Thanks.

**Comment by huanrwan-amd** [comment]
This issue is tracked in amd-smi: https://github.com/ROCm/amdsmi/issues/38 .

**Comment by huanrwan-amd** [comment, marked as answer]
Hi @parthraut,
There are changes for replacing "power" to "energy_accumulator" in https://github.com/ROCm/amdsmi/commit/95ca2b83a10b5066176b98d25c81a90e9317515d#diff-377974e1be0b08ba3dbdaf301b7687aa2b3b754ef03d6cc3120f22f52192282fR1535

In the mean time, you could use amdsmi_get_power_info() for current power consumptions: https://rocm.docs.amd.com/projects/amdsmi/en/latest/reference/amdsmi-py-api.html#amdsmi-get-power-info
